pkg/doc/jwt/verifier.go

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright SecureKey Technologies Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package jwt
+
+import (
+	"crypto"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+)
+
+// Verifier makes verification of JSON Web Token.
+type Verifier interface {
+
+	// Verify verifies JWT.
+	Verify(j JSONWebToken) error
+}
+
+// KeyResolver resolves public key based on what and kid.
+type KeyResolver interface {
+
+	// Resolve resolves public key.
+	Resolve(what, kid string) (interface{}, error)
+}
+
+type basicVerifier struct {
+	resolver KeyResolver
+}
+
+// NewVerifier creates a new basic Verifier.
+func NewVerifier(resolver KeyResolver) Verifier {
+	return &basicVerifier{resolver: resolver}
+}
+
+// Verify verifies JSON Web Token.
+func (v *basicVerifier) Verify(j JSONWebToken) error {
+	claims := Claims{}
+
+	err := j.Claims(&claims)
+	if err != nil {
+		return fmt.Errorf("read claims from JSON Web Token: %w", err)
+	}
+
+	kid := j.HeaderString(HeaderKeyID)
+
+	pubKey, err := v.resolver.Resolve(claims.Issuer, kid)
+	if err != nil {
+		return err
+	}
+
+	alg := j.HeaderString(HeaderAlgorithm)
+	if alg == "" {
+		return errors.New("JWS alg is not defined")
+	}
+
+	verificationData, err := j.VerificationData()
+	if err != nil {
+		return fmt.Errorf("build JWS verification data: %w", err)
+	}
+
+	switch alg {
```

Comment:
TODO: the verifiers should be registered against alg in the constructor (allowing for default implementation).

Reply:
I was thinking about something similar! Added a reference to your comment to https://github.com/hyperledger/aries-framework-go/issues/1267.